esp/services/ws_workunits/ws_wudetails.cpp

```diff
@@ -70,36 +76,39 @@ class WUDetailsVisitor : public IWuScopeVisitor
     public:
         void add(WuAttr attr, const char *value)
         {
-            AttribValuePair tmp(attr,value);
-            attribValuePairs.insert(tmp);
+            AttribValuePair tmp(attr,"");
+
+            auto cur = find(attribValuePairs.begin(), attribValuePairs.end(), tmp);
+            if (cur!=attribValuePairs.end())
+            {
+                StringBuffer newval (cur->second.str());
+                newval.append(", ");
+                newval.appendf("\"%s\"",  value);
+                auto pos = distance(attribValuePairs.begin(), cur);
+                attribValuePairs.at(pos) = AttribValuePair(attr, newval);
+            }
+            else
+            {
+                StringBuffer newval ("[");
+                newval.appendf("\"%s\"",  value);
+                AttribValuePair newvalpair(attr, newval);
```

Comment:
For a little bit more efficiency you can use emplace_back avoiding the local object.

Reply:
You could alternatively append the value to temp.second and insert that.